src/pages/List.js

```diff
@@ -42,38 +56,44 @@ const List = props => {
         // isLoading = is a Boolean that represents the loading status for the firebase query. true until an initial payload from Firestore is received.
         // data = an Array containing all of the documents in the collection. Each item will contain an id along with the other data contained in the document.
         render={({ isLoading, data }) => {
-          // if (!isLoading && data.length === 0) {
-          //   return <ErrorMessage />;
-          // }
-
           if (!isLoading) {
             setShoppingList(data);
           }
-
           return isLoading ? (
             <Loading />
           ) : (
-            <ul className="shopping-list">
-              {shoppingList.map((item, index) => (
-                <li key={index}>
-                  <label>
-                    <input
-                      type="checkbox"
```

Comment:
@stevelikesmusic I'm getting a bug with our checkboxes where for *some* I have to double click before the UI updates. Do you know what might cause that?

Reply:
I haven't been able to reproduce the bug locally @prophen 😖

Reply:
I see it now. I think this is a result of how we're getting data back from the `FirebaseCollection` component and then saving in the context component. When I debug, we get the correct purchase date for the checked item. And then when we save the list into context, the next render has the purchase date as `null`.

I think the fix would be what I mentioned before—use one or the other (Context or `FirebaseCollection`) to fetch/hold data.